hugr/src/types/poly_func.rs

```diff
@@ -73,9 +73,12 @@ impl PolyFuncType {
 
     /// Validates this instance, checking that the types in the body are
     /// wellformed with respect to the registry, and the type variables declared.
-    pub fn validate(&self, reg: &ExtensionRegistry) -> Result<(), SignatureError> {
+    /// Allows both inputs and outputs to contain [RowVariable]s
+    ///
+    /// [RowVariable]: [crate::types::TypeEnum::RowVariable]
+    pub fn validate_varargs(&self, reg: &ExtensionRegistry) -> Result<(), SignatureError> {
```

Comment:
This is a stranger case than [TypeRow](https://github.com/CQCL/hugr/pull/804/files/7214cbc328f0991e5babc0d9be2144d9050619b4#r1599919712). We *do* have a use for a PolyFuncType that doesn't allow row vars: the signature of a FuncDecl/FuncDefn (i.e. inside the binders). However, it happens that we never validate that, and I'm much less sure why we don't. Hmmm.

If we can distinguish Signature from FunctionType, then we might be able to distinguish the two variants of PolyFuncType, too. Lemme see....

Reply:
Yah you could never link in a single FuncDefn, but you could replace every Call to it with a different Call (with some typeargs missing, and targetting a different FuncDefn with those typeargs "fixed").

Is that really useful? I feel a bit unconvinced. Maybe one should just have different FuncDecls to start with?

Reply:
But the handling of FuncDecl's is only really relevant if we want to go with branch `feat/row_typevar4` (splitting FunctionType and PolyFuncType), which I don't think we should do here. (As a second step if we want to at all.)

I will stick by not using `validate` - even if `validate_fixed_len` is not defined here, it would be equally valid (just that we don't use it). Happy to entertain any non-ambiguous alternative to `validate_varargs` tho.